### DISCUSSION TITLE
Using the same loader on multiple modules causes error

### DISCUSSION THREAD

**Discussion by Jakobud** [webpack/webpack#12972] [Q&A] [answered]
I have a situation where I have multiple modules using the same loader. Here is a really simple example:

```
...
module: {
  rules: [
    {
      test: /(.)*\.scss$/,
      resourceQuery: 'website1',
      user: [
        ...
        { loader: MiniCssExtractPlugin.loader },
        ...
      ]
    },
    {
      test: /(.)*\.scss$/,
      resourceQuery: 'website2',
      user: [
        ...
        { loader: MiniCssExtractPlugin.loader },
        ...
      ]
    },
    ...
```
I have a kinda weird setup where I have the same entry files being processed and bundled multiple times for multiple websites depending on the `resourceQuery` on the entry file.

Anyways I get this error:

```
ModuleBuildError: Module build failed (from ./node_modules/mini-css-extract-plugin/dist/loader.js):
Error: Didn't get a result from child compiler
```

After a little research I found this:
https://github.com/webpack-contrib/mini-css-extract-plugin/issues/126

This thread implies that you get this error because I'm using the same loader for 2 different modules. Is that right? I also have a similar setup as above but for compiling my JS and I use the same loaders for that on multiple modules in the same way and it works just fine.

Is there a solution for this? Is this a Webpack problem or MiniCssExtractPlugin problem?

**Comment by alexander-akait** [comment, marked as answer]
Please provide reproducible test repo

**Reply by Jakobud** [reply]
I will get back to you soon on this. I'm trying to break down my config into something really simple and reproducible. Thanks for your support.

**Reply by Jakobud** [reply]
You can mark this solve I figured it out. My resourceQuery test rules were not formatted correctly which led to the same loaders being used in duplicate of the exact same rules. Got it working though. Thank you,
